url9ecf1e8297_nhamidi_ninja_tgz-pJ8-net_fec_openrq_test_TestRunnerJ8/src/Recepteur_thread.java

```diff
@@ -0,0 +1,62 @@
+
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.SocketException;
+
+public class Recepteur_thread extends Thread {
+	private int port;
+	private DatagramSocket socketReception;
+	private boolean end_loop;
+	byte[] packetData;
+	private int number_of_packets;
+
+	public Recepteur_thread(int port) throws SocketException {
+		this.port = port;
+		this.end_loop = true;
+		this.socketReception = new DatagramSocket(this.port);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *socketReception*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).